utilities/write_batch_with_index/write_batch_with_index.cc

```diff
@@ -172,6 +233,18 @@ class BaseDeltaIterator : public Iterator {
     return delta_iterator_->status();
   }
 
+  bool ChecksLowerBound() const override { return false; }
```

Comment:
Do you mean "return true"? If so, I still question why we need the ChecksLower/UpperBound functions.

If you do mean "return false," can you explain?

Reply:
I do mean false. This PR is about implementing support for the upper_bound, much of the machinery is there for the lower_bound because it is common, but my focus was implementing for the upper_bound.